tests/lib/openstack_query/api/test_query_api.py

```diff
@@ -40,14 +43,19 @@ def _run_with_test_case(data_subset, mock_kwargs):
         )
 
         # test that data is marshalled correctly to executer
-        assert (
-            instance.executer.client_side_filter_func
-            == instance.builder.client_side_filter
-        )
-        assert (
-            instance.executer.server_side_filters
-            == instance.builder.server_side_filters
-        )
+        # - this differs based on if from_subset is given
+        if data_subset:
+            client_filters = (
+                instance.builder.client_side_filters
+                + instance.builder.server_filter_fallback
+            )
+            server_filters = None
+        else:
+            client_filters = instance.builder.client_side_filters
+            server_filters = instance.builder.server_side_filters
```

Comment:
This if-else should be two different test cases
If this is a PITA to test where we don't want to write new cases then it's a bad code smell